boards/opentitan/src/tests/flash.rs

```diff
@@ -2,474 +2,417 @@
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 // Copyright Tock Contributors 2022.
 
-//! Test the opentitan flash implementation
-
-mod key_value {
-    use crate::tests::run_kernel_op;
-    use crate::{SIPHASH, TICKV};
-    use capsules_core::virtualizers::virtual_flash::FlashUser;
-    use capsules_extra::test::kv_system::KVSystemTest;
-    use capsules_extra::tickv::KVSystem;
-    use capsules_extra::tickv::{TicKVKeyType, TicKVSystem};
-    use kernel::debug;
-    use kernel::hil::hasher::Hasher;
-    use kernel::static_init;
-    use kernel::utilities::leasable_buffer::SubSliceMut;
-
-    #[test_case]
-    fn tickv_append_key() {
-        debug!("start TicKV append key test...");
-
-        unsafe {
-            let tickv = TICKV.unwrap();
-            let sip_hasher = SIPHASH.unwrap();
-
-            let key_input = static_init!(
-                [u8; 16],
-                [
-                    0x12, 0x34, 0x56, 0x78, 0x9A, 0xBC, 0xDE, 0xF0, 0x12, 0x34, 0x56, 0x78, 0x9A,
-                    0xBC, 0xDE, 0xF0
-                ]
-            );
-            let key = static_init!([u8; 8], [0; 8]);
-            let value = static_init!([u8; 3], [0x10, 0x20, 0x30]);
-            let ret = static_init!([u8; 4], [0; 4]);
```

Comment:
Why did you remove test cases? Does this PR cause a regression but instead of fixing it you removed the tests?

Reply:
Pretty sure I didn't remove them. @bradjc may confirm this.

I didn't perform any tests on actual hardware so I could not be aware of any regression tests. Initially, I only changed the signature of flash methods and a small detail design implementation. However, the detail design was rejected by @bradjc , so there was not any change in code logic.

The sole explanation that I have: it looks like my pull request was merged on 10th October, while yours on 9th October, so your work was overwritten by my pull request. Locally, I still have `tick_test.rs` file that you removed. I apologize for my mistake!

I can fix this, but I would require someone to run the test suite on OpenTitan. What do you think?

Reply:
> The sole explanation that I have: it looks like my pull request was merged on 10th October, while yours on 9th October, so your work was overwritten by my pull request. Locally, I still have `tickv_test.rs` file that you removed. I apologize for my mistake!

Urgh. I would hope that would have been caught by review to ensure we don't get nasty merge conflicts like this.

That's ok! I was just confused why test cases are being removed